### PR TITLE
Add `java.lang.Double` to RJMX serialization configuration

### PR DESCRIPTION
**Pull request by roberttoyonaga**
Related Issue: https://github.com/oracle/graal/issues/9349

The OS platform MXBean uses double values for some attributes. See [N/A] fields below:
image

Adding serialization config for java.lang.Double fixes this problem.

### CODE REVIEW

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxCommonFeature.java [hunk 1]
```diff
@@ -207,6 +207,7 @@ private static void configureSerialization(BeforeAnalysisAccess access) {
                         "[B", "com.oracle.svm.core.jdk.UnsupportedFeatureError",
                         "java.io.IOException", "java.lang.Boolean", "java.lang.ClassCastException", "java.lang.Error",
                         "java.lang.Exception", "java.lang.IllegalArgumentException", "java.lang.IllegalStateException",
+                        "java.lang.Double",
```
Comment (fniephaus) [inline substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxCommonFeature.java:210]: Should we, for consistency, also add `[Ljava.lang.Double;`, and other `java.lang` base types like `Character` and `Float` (or are these never used in JMX)?

Comment (roberttoyonaga) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxCommonFeature.java:210]: It doesn't look like any of the platform mxbean attributes use them, but it's possible that some user's custom standard mbeans may use them. So we might as well add them to the serialization config too.

**Comment by roberttoyonaga** [comment]
GHA failures:

- GraalVM Gate / /compiler style,fullbuild,test JDK21 
- GraalVM Gate / /substratevm on Windows

Both failures seem unrelated to these changes. They are also failing on other PRs too.

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxCommonFeature.java (hunk 1 above)
Comment (fniephaus) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxCommonFeature.java]: If that's the case, I'd argue that the user also needs to provide custom serialization config. Could you just revert to the previous version again?

**Comment by fniephaus** [comment]
>Both failures seem unrelated to these changes. They are also failing on other PRs too.

yes, we are aware. don't worry about them :)

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxCommonFeature.java (hunk 1 above)
Comment (roberttoyonaga) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxCommonFeature.java]: ok will do

**Review by fniephaus — APPROVED** [review]
LGTM

**Comment by fniephaus** [comment]
This got merged via https://github.com/oracle/graal/commit/ba1f2656418eceef438d452e88d0aa2a96f2087c (we only picked https://github.com/oracle/graal/commit/123e621e8f7a7a325e2e7f2bd876a46db4708407, and remove the other commits that you reverted).

Thanks for fixing this, Robert!

**Comment by ansalond** [comment]
Also backported to the release branch of the next GraalVM 24.1 release:
https://github.com/oracle/graal/commit/79d2f9a2ebf6aab27fbc693be38149112c106c1e

**Pull request closed without merging**